arcgis-ios-sdk-samples/Maps/Show location history/LocationHistoryViewController.swift

```diff
@@ -0,0 +1,191 @@
+
+        overlay.renderer = AGSSimpleRenderer(symbol: locationSymbol)
+
+        return overlay
+    }()
+    
+    private let trackOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+
+        let trackSymbol = AGSSimpleLineSymbol(style: .solid, color: Constants.trackColor, width: Constants.trackWidth)
+        overlay.renderer = AGSSimpleRenderer(symbol: trackSymbol)
+
+        return overlay
+    }()
+
+    private var trackBuilder: AGSPolylineBuilder?
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupView()
+    }
+    
+    // MARK: IBActions
+    
+    @IBAction private func trackingTapped(_ sender: UIBarButtonItem) {
+        toggleTrackingStatus()
+    }
+    
+    // MARK: Private behavior
+    
+    private func handleLocationStatusChange() {
+        switch currentStatus {
+        case .paused:
+            stopProcessingLocationChanges()
+        case .inProgress:
+            startProcessingLocationChanges()
+        }
+        
+        updateView(status: currentStatus)
+    }
+    
+    private func processLocationUpdate() {
+        guard currentStatus == .inProgress, let position = mapView.locationDisplay.mapLocation, position.x != 0, position.y != 0 else { return }
+        
+        let locationGraphic = AGSGraphic(geometry: position, symbol: nil)
+        locationsOverlay.graphics.add(locationGraphic)
+
+        guard let trackBuilder = trackBuilder else { return }
+
+        trackBuilder.add(position)
+        let trackGraphic = AGSGraphic(geometry: trackBuilder.toGeometry(), symbol: nil)
+        trackOverlay.graphics.add(trackGraphic)
+    }
+    
+    private func setupLocationDisplay(_ locationDisplay: AGSLocationDisplay) {
+        locationDisplay.autoPanMode = .recenter
+        locationDisplay.initialZoomScale = Constants.initialZoomScale
+        
+        locationDisplay.start { (error) in
+            if let error = error {
+                print("Error starting location display: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    private func setupMapView() {
+        mapView.map = map
+
+        map.load(completion: { [weak self, unowned map] (error) in
+            self?.trackBuilder = AGSPolylineBuilder(spatialReference: map.spatialReference)
+        })
+
+        mapView.graphicsOverlays.addObjects(from: [trackOverlay, locationsOverlay])
+
+        setupLocationDisplay(mapView.locationDisplay)
+
+        updateView(status: currentStatus)
+    }
+    
+    private func setupNavigationBar() {
+        guard let sourceBarButtonItem = navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem else {
+            return
+        }
+
+        sourceBarButtonItem.filenames = ["LocationHistoryViewController"]
+    }
+    
+    private func setupView() {
```

Comment:
This is only called form `viewDidLoad()`, so it might make sense to just inline the contents there. After all, the documentation basically says that's what it is for: "You usually override this method to perform additional initialization on views that were loaded from nib files."

Reply:
Addressed in [`524f91ca`](https://github.com/Esri/arcgis-runtime-samples-ios/pull/751/commits/524f91ca1566a4f9811c86bf190b053bdc29ec0a).